File: atcoder/go/abc144/d.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var a, b, x float64
	fmt.Scan(&a, &b, &x)
	if x > a*a*b/2.0 {
		var tmp float64 = math.Atan(2.0 * (a*a*b - x) / (a * a * a))
		fmt.Println(tmp * 180 / math.Pi)
	} else {
		var tmp float64 = math.Atan(a * b * b / x / 2.0)
		fmt.Println(tmp * 180 / math.Pi)
	}
}

File: atcoder/python/abc144/d.py
import math
a,b,x=map(int,input().split())
lim=a*a*b/2
if x<=lim:
    rad=math.atan(a*b*b/(2*x))
    ans=math.degrees(rad)
    print(ans)
else:
    rad=math.atan(2*(a*a*b-x)/(a**3))
    ans=math.degrees(rad)
    print(ans)